### PR TITLE
Add settings to preferences panel

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

Add the following settings to preferences panel. Serialize user preference.

https://jira.autodesk.com/browse/DYN-5203

preference settings

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes
Add settings to preferences panel

### Reviewers
@QilongTang @jesusalvino @Amoursol

### CODE REVIEW

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs [hunk 1]
```diff
@@ -433,6 +436,34 @@ private void exportTextBlock_MouseLeftButtonDown(object sender, MouseButtonEvent
...
+            var args = e as PythonTemplatePathEventArgs;
+            args.Cancel = true;
+
+            var dialog = new System.Windows.Forms.OpenFileDialog
+            {
+                // Navigate to initial folder.
+                FileName = args.Path
+            };
+
+            dialog.Filter = "All Files (*.*)|*.*";
```
Comment (reddyashish) [inline src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs:453]: Do you think we should be limiting this to only files with .py extension?

Comment (Amoursol) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs:453]: @reddyashish agreed - if this is specifically for Python we should explicitly limit it to that 😄

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs [hunk 2]
```diff
@@ -61,6 +62,8 @@ public PreferencesView(DynamoView dynamoView)
             //We need to store the ScaleFactor value in a temporary variable always when the Preferences dialog is created.
             scaleValue = dynViewModel.ScaleFactorLog;
             ResetGroupStyleForm();
+
+            viewModel.RequestShowFileDialog += OnRequestShowFileDialog;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs:66]: unsubscribe this somewhere?

**Comment by QilongTang** [comment]
Can we append some tests to serialize the new settings? I think we already have a preference setting somewhere so you only need to touch the settings.xml and update the test.

**Comment by reddyashish** [comment]
Yes, looking into the tests now.

**Comment by QilongTang** [comment]
Oops, there is some conflict to address with https://github.com/DynamoDS/Dynamo/pull/13355

**Comment by reddyashish** [comment]
@QilongTang Added the BackupInterval and HardwareAcceleration property to the preferences-serialization tests. MaxNumRecentFiles is already added to this test and we have 2 separate tests for the loading of PythonTemplate. We also have a CanUpdatePythonTemplateSettings() test in the CoreTests class that checks: changing the PythonTemplate. So that should cover it.

**Review by QilongTang — APPROVED** [review]

**Pull request merged**